app/types/scheduler.py

```diff
@@ -0,0 +1,229 @@
+import asyncio
+import logging
+from collections.abc import AsyncGenerator, Callable, Coroutine
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING, Any
+
+from arq.connections import RedisSettings
+from arq.jobs import Job
+from arq.typing import WorkerSettingsBase
+from arq.worker import create_worker
+from sqlalchemy.ext.asyncio import AsyncSession
+
+if TYPE_CHECKING:
+    from arq import Worker
+
+scheduler_logger = logging.getLogger("scheduler")
+
+
+async def run_task(
+    ctx,
+    job_function: Callable[..., Coroutine[Any, Any, Any]],
+    _get_db: Callable[[], AsyncGenerator[AsyncSession, None]],
+    **kwargs,
+):
```

Comment:
What happens if this method raises an error? 

If the exception is caught by Arq (I hope and believe this should be the case), we should wrap the body of this method in a try/except statement. If an exception occurs, we want to log the exception trace, maybe the job id, and reraise the exception to let Arq know the job failed.